chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -800,10 +795,10 @@ impl PeerManagerActor {
             PeerActor::add_stream(
                 ThrottledFrameRead::new(read, Codec::default(), rate_limiter.clone(), semaphore)
                     .take_while(|x| match x {
-                        Ok(_) => future::ready(true),
+                        Ok(_) => true,
                         Err(e) => {
                             warn!(target: "network", ?e, "Peer stream error");
-                            future::ready(false)
+                            false
                         }
                     })
                     .map(Result::unwrap),
```

Comment:
One of those rare cases where there *should* be a better way to do this, but there isn't one currently. Opened https://github.com/tokio-rs/tokio/issues/4337.

Reply:
btw, this was fixed upstream

Reply:
@matklad Thanks. We can make it even better in the future.